velox/functions/prestosql/aggregates/CovarianceAggregates.cpp

```diff
@@ -355,38 +440,44 @@ class RegrIntermediateInput : public CovarIntermediateInput {
  public:
   explicit RegrIntermediateInput(const RowVector* rowVector)
       : CovarIntermediateInput(rowVector, kRegrIndices),
-        m2X_{asSimpleVector<double>(rowVector, kRegrIndices.m2X)} {}
+        m2X_{asSimpleVector<double>(rowVector, kRegrIndices.m2X)},
+        m2Y_{asSimpleVector<double>(rowVector, kRegrIndices.m2Y)} {};
 
   void mergeInto(RegrAccumulator& accumulator, vector_size_t row) {
     accumulator.merge(
         count_->valueAt(row),
         meanX_->valueAt(row),
         meanY_->valueAt(row),
         c2_->valueAt(row),
-        m2X_->valueAt(row));
+        m2X_->valueAt(row),
+        m2Y_->valueAt(row));
   }
 
  private:
   SimpleVector<double>* m2X_;
+  SimpleVector<double>* m2Y_;
 };
 
 class RegrIntermediateResult : public CovarIntermediateResult {
  public:
   explicit RegrIntermediateResult(const RowVector* rowVector)
       : CovarIntermediateResult(rowVector, kRegrIndices),
-        m2X_{mutableRawValues<double>(rowVector, kRegrIndices.m2X)} {}
+        m2X_{mutableRawValues<double>(rowVector, kRegrIndices.m2X)},
+        m2Y_{mutableRawValues<double>(rowVector, kRegrIndices.m2Y)} {};
 
   static std::string type() {
-    return "row(double,bigint,double,double,double)";
+    return "row(double,bigint,double,double,double,double)";
```

Comment:
Does that mean the intermediate result type of existing function is changing?

We have been discussing that intermediate results are also part of public API and need to be backwards compatible. Some systems (streaming) are storing intermediate results on disk and they break if these results become non-backwards compatible.

CC: @kgpai @bikramSingh91 @Yuhta

Reply:
Understood. Instead of directly modifying the intermediate results in the class RegrIntermediateResult, I declare a new class representing the intermediate type by inheriting the class RegrIntermediateResult.